flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -652,13 +674,15 @@ public void setNumBuffers(int numBuffers) {
                 // one buffer from NetworkBufferPool
                 return;
             }
-
-            if (checkAvailability()) {
+            AvailabilityStatus availabilityAndRequestFromGlobalPool = checkAvailability();
+            if (availabilityAndRequestFromGlobalPool.isAvailable()) {
                 toNotify = availabilityHelper.getUnavailableToResetAvailable();
             } else {
                 availabilityHelper.resetUnavailable();
             }
-
+            if (availabilityAndRequestFromGlobalPool.isNeedRequestFromGlobalWhenAvailable()) {
+                requestMemorySegmentFromGlobalWhenAvailable();
+            }
             checkConsistentAvailability();
```

Comment:
This common pattern has repeated 4 times. It would be nice to deduplicate it.

Reply:
good suggestion.